source/validate.cpp

```diff
@@ -177,9 +179,12 @@ spv_result_t ProcessInstruction(void* user_data,
   // The IdPass check registers instructions and, therefore, must be called
   // before any instruction lookups are performed.
   if (auto error = IdPass(_, inst)) return error;
+
+  const Instruction* instruction = &(_.ordered_instructions().back());
+
   if (auto error = DataRulesPass(_, inst)) return error;
   if (auto error = ModuleLayoutPass(_, inst)) return error;
-  if (auto error = CfgPass(_, inst)) return error;
+  if (auto error = CfgPass(_, instruction)) return error;
```

Comment:
I think this can be propagated through the other passes as well and then we always provide the higher level Instruction instead of the spv_parsed_instruction_t.

I didn't do it here to keep the change smaller.